osbenchmark/builder/builder.py

```diff
@@ -62,6 +62,14 @@ def install(cfg):
     master_nodes = cfg.opts("builder", "master.nodes")
     seed_hosts = cfg.opts("builder", "seed.hosts")
 
+    # Ensure node_name and master_nodes match, using node_name as the default
+    if node_name not in master_nodes:
+        print(
+            f"The provided --node-name '{node_name}' and --master-nodes '{master_nodes}' are different. "
+            f"Using '{node_name}' for both node name and initial master node."
+        )
+        master_nodes = [node_name]
```

Comment:
I am wondering, If this is the desired behavior then do we want to expose master_nodes ? Here we are reporting error but override the master_nodes values and continue with provisioning. I think we should either fail so that user is notified or not expose the master_nodes field at all. @IanHoang would like to know your thoughts as well